ccdproc/core.py

```diff
@@ -1558,6 +1558,137 @@ def cosmicray_median(ccd, error_image=None, thresh=5, mbox=11, gbox=0,
+    a specified sigma threshold factor times the local sigma above and below
+    zero (the lsigma and hsigma parameters). This is done for individual pixels
+    and then for column sums of pixels (excluding previously flagged bad pixels)
+    from two to the number of lines in the image. The sigma of the sums is
+    scaled by the square root of the number of pixels summed so that
+    statistically low or high column regions may be detected even though
+    individual pixels may not be statistically deviant. For the purpose of this
+    task one would normally select large sigma threshold factors such as six or
+    greater to detect only true bad pixels and not the extremes of the noise
+    distribution.
+
+    As a final step each column is examined to see if there are small segments
+    of unflagged pixels between bad pixels. If the length of a segment is less
+    than that given by the ngood parameter all the pixels in the segment are
+    also marked as bad.
+
+
+    Returns
+    -------
+    mask : `numpy.ndarray`
+        A boolean ndarray with the bad pixel positions identified.
+    '''
+    mask = ~np.isfinite(ratio.data)
+    medsub = ratio.data - ndimage.filters.median_filter(ratio.data,
+                                                        size=(nlsig, ncsig))
+
+    nl, nc = ratio.data.shape
+    nbl = int(np.ceil(ratio.data.shape[0] / nlsig))
+    nbc = int(np.ceil(ratio.data.shape[1] / ncsig))
+    for i in range(nbl):
+        for j in range(nbc):
+            l1 = i*nlsig
+            l2 = (i+1)*nlsig
+            if l2 > nl:
+                l2 = nl
+            c1 = j*ncsig
+            c2 = (j+1)*ncsig
+            if c2 > nc:
+                c2 = nc
+            block = medsub[l1:l2,c1:c2]
+            high = np.percentile(block.ravel(), 69.1)
+            low = np.percentile(block.ravel(), 30.9)
+            block_sigma = (high-low)/2.0
+            block_mask = ( (block > hsigma*block_sigma) |
+                           (block < -lsigma*block_sigma) )
+            mblock = np.ma.MaskedArray(block, mask=block_mask)
+
+            if findbadcolumns:
+                csum = np.ma.sum(mblock, axis=0)
+                csum_sigma = np.ma.MaskedArray([np.sqrt(c2-c1-x)*block_sigma
+                                   if c2-c1-x > 0 else 0
+                                   for x in np.ma.sum(mblock.mask, axis=0)])
+                colmask = ( (csum.filled(1) > hsigma*csum_sigma) |
+                            (csum.filled(1)< -lsigma*csum_sigma) )
+                for c in range(c2-c1):
+                    block_mask[:,c] = block_mask[:,c] | np.array([colmask[c]]*(l2-l1))
+
+            mask[l1:l2,c1:c2] = block_mask
+
+    if findbadcolumns:
+        for c in range(0,nc):
+            for l in range(0,nl-ngood-1):
+                if mask[l,c] == True:
+                    for i in range(2,ngood+2):
+                        if (mask[l+i,c] == True) and not np.all(mask[l:l+i+1,c]):
```

Comment:
`l+i` is computed several times here without knowing what it indicates. Could you do the computation of it as seperate statement (performance) and give it a meaningful variable name?